### PR TITLE
Squash app source user bug

### LINKED ISSUE #194: Source User Crash
Ever since adding the code to translate source user to username and display name in metadata, legobot will crash if:
-Slack Connector
-Message is an attachment

### PR DESCRIPTION
**Pull request by drewpearce**
Resolves #194.

This branch includes release prep (changes.md and setup.py)
Once merged we should be good for a release (1.2.3)

### CODE REVIEW

File: Legobot/Connectors/Slack.py [hunk 1]
```diff
@@ -170,8 +170,10 @@ def _parse_metadata(self, message):
         elif 'bot_id' in message:
             metadata['source_user'] = message['bot_id']
 
-        metadata['user_id'] = metadata['source_user']
-        metadata['display_name'] = self.get_username(metadata['source_user'])
+        if 'source_user' in metadata:
```
Comment (bbriggs) [inline Legobot/Connectors/Slack.py]: What happens if a lego depends on `source_user`?

Comment (drewpearce) [reply Legobot/Connectors/Slack.py]: it's out of luck?
to be clear we weren't accommodating for this use case previously, simply ignoring it. With the addition of trying to convert source_user to these other metadata pieces, the missing source_user caused the bot to crash. This is an attempt to return to status quo (ignoring) not to accommodate the odd messages. If you want to kick it back I'll look for a full remedy.

Comment (bbriggs) [reply Legobot/Connectors/Slack.py]: Base fields should be a `None` when they can't be filled, so we should add this at the parent Lego level. The idea is that this is part of the standard set of metadata and we don't want legos to throw a key error when looking for it.

**Comment by bbriggs** [comment]
It turns out APIs are hard. Who knew?

**Pull request merged**